frontend/src/types.ts

```diff
@@ -1500,6 +1505,16 @@ export interface PropertyGroupFilterValue {
     values: AnyPropertyFilter[]
 }
 
+export interface CohortCriteriaGroupFilter {
+    type: FilterLogicalOperator
+    values: CohortCriteriaGroupFilterValue[]
+}
+
+export interface CohortCriteriaGroupFilterValue {
```

Comment:
It's confusing to see both `CohortCriteriaGroupFilter` and `CohortCriteriaGroupFilterValue`.

It looks like the intent is to create some sort of recursive structure, probably to chain logical operators. If that's the case, it could be implemented as,

```
export interface CohortFilter {
  type: FilterOperator;
  values: CohortGroup | Partial<CohortGroup> | CohortFilter; // Self referential.
}
```

Unless I'm missing something.

(I trimmed down some words I thought weren't adding much value, but maybe they are. I don't know much about the domain.)

Reply:
That's awesome didn't know that was possible!